burn-core/src/record/tensor.rs

```diff
@@ -128,7 +128,7 @@ impl<B: Backend, const D: usize> Record for Tensor<B, D, Bool> {
         BoolTensorSerde::new(self.into_data().serialize())
     }
 
-    fn from_item<S: PrecisionSettings>(item: Self::Item<S>) -> Self {
+    fn from_item<S: PrecisionSettings>(item: Self::Item<S>, _device: &B::Device) -> Self {
         Tensor::from_data(item.data, &B::Device::default())
```

Comment:
Shouldn't device be passed instead of `&B::Device::default()`?

Reply:
Yes, thanks for spotting that mistake 🙏